module/zfs/dbuf.c

```diff
@@ -48,6 +48,71 @@
 #include <sys/callb.h>
 #include <sys/abd.h>
 
+kstat_t *dbuf_ksp;
+
+typedef struct dbuf_stats {
+	kstat_named_t dbuf_cache_count;
+	kstat_named_t dbuf_cache_size;
+	kstat_named_t dbuf_cache_size_max;
+	kstat_named_t dbuf_cache_max_bytes;
+	kstat_named_t dbuf_cache_lowater_bytes;
+	kstat_named_t dbuf_cache_hiwater_bytes;
+	kstat_named_t dbuf_cache_total_evicts;
+	kstat_named_t dbuf_cache_levels[DN_MAX_LEVELS];
+	kstat_named_t dbuf_cache_levels_bytes[DN_MAX_LEVELS];
+	kstat_named_t hash_hits;
+	kstat_named_t hash_misses;
+	kstat_named_t hash_collisions;
+	kstat_named_t hash_elements;
+	kstat_named_t hash_elements_max;
+	kstat_named_t hash_chains;
+	kstat_named_t hash_chain_max;
+	kstat_named_t hash_insert_race;
+	kstat_named_t hash_dbuf_levels[DN_MAX_LEVELS];
+	kstat_named_t hash_dbuf_levels_bytes[DN_MAX_LEVELS];
+} dbuf_stats_t;
+
+dbuf_stats_t dbuf_stats = {
+	{ "dbuf_cache_count",			KSTAT_DATA_UINT64 },
+	{ "dbuf_cache_size",			KSTAT_DATA_UINT64 },
+	{ "dbuf_cache_size_max",		KSTAT_DATA_UINT64 },
+	{ "dbuf_cache_max_bytes",		KSTAT_DATA_UINT64 },
+	{ "dbuf_cache_lowater_bytes",		KSTAT_DATA_UINT64 },
+	{ "dbuf_cache_hiwater_bytes",		KSTAT_DATA_UINT64 },
+	{ "dbuf_cache_total_evicts",		KSTAT_DATA_UINT64 },
+	{ { "dbuf_cache_levels_N",		KSTAT_DATA_UINT64 } },
+	{ { "dbuf_cache_levels_bytes_N",	KSTAT_DATA_UINT64 } },
+	{ "hash_hits",				KSTAT_DATA_UINT64 },
+	{ "hash_misses",			KSTAT_DATA_UINT64 },
+	{ "hash_collisions",			KSTAT_DATA_UINT64 },
+	{ "hash_elements",			KSTAT_DATA_UINT64 },
+	{ "hash_elements_max",			KSTAT_DATA_UINT64 },
+	{ "hash_chains",			KSTAT_DATA_UINT64 },
+	{ "hash_chain_max",			KSTAT_DATA_UINT64 },
+	{ "hash_insert_race",			KSTAT_DATA_UINT64 },
+	{ { "hash_dbuf_levels_N",		KSTAT_DATA_UINT64 } },
+	{ { "hash_dbuf_levels_bytes_N",		KSTAT_DATA_UINT64 } }
+};
+
+#define	DBUF_STAT_SET(stat, val)	\
+	(dbuf_stats.stat.value.ui64 = (val));
+#define	DBUF_STAT_INCR(stat, val)	\
+	atomic_add_64(&dbuf_stats.stat.value.ui64, (val));
+#define	DBUF_STAT_DECR(stat, val)	\
+	DBUF_STAT_INCR(stat, -(val));
+#define	DBUF_STAT_BUMP(stat)		\
+	DBUF_STAT_INCR(stat, 1);
+#define	DBUF_STAT_BUMPDOWN(stat)	\
+	DBUF_STAT_INCR(stat, -1);
+#define	DBUF_STAT_MAX(stat, val) {					\
```

Comment:
Please make DBUF_STAT_MAX into a function since you're declaring variables.

Reply:
Renamed the variable to be `__m` so it's less likely to have a collision.